ocsci/framework.py

```diff
@@ -141,4 +144,9 @@ def _teardown(self):
             # TODO: update post_to_polarion to look for correct keys
         self.end = time.time()
         self.duration = self.end - self.start
+
+        if self.destroy_cluster:
```

Comment:
Do I understand that it will be called after deployment finish in its teardown and if any other test is in test suite it cannot succeed cause we will destroy cluster before continue to next test case?

Reply:
Yes you are correct if those subsequent tests do not create a new cluster, so there is some responsibility on those creating the suite to not opt into destroying their cluster before they perform tests on it. That's why this defaults to False.

Reply:
Yep, but I was thinking that better approach would be to allow user to do this destroy cluster at the end of the execution. If I understand correctly now there are two options:

1) set destroy-cluster in test suite yaml file for particular test case (user will specify to last one I guess), if they will add one test to test suite they will need to move this param to new last test case.
2) do the destroy cluster at the start of the execution with --destroy parameter

But what if user would like to run whole test suite and decide not to destroy at the end. If destroy is defined in test suite yaml file, there is currently no way to skip destroy right?

So if we would like to support this, we probably will have to have two params. --destroy and --no-destroy or --no-destroy and destroy by default and the end of execution and --destroy-at-start to call the destroy at the start as you do.

Reply:
I think your approach of having the destroy be it's own test would be better. This makes it less clunky to add additional tests into an established suite. Also providing a `--no-destroy` option that the test could read in and return TestStatus.SKIPPED in the event of would be nice. This would allow users to leave their clusters up without having to manually edit the suite file before running, as well as let us trigger this through jenkins more easily.

Reply:
One thing to add to this however is that if the destroy is it's own test, and we hit issues during our installation, we have the test set to 'abort-on-fail.' This will stop us from getting to the destroy test as we stop test execution. Just something to think about, and a tradeoff if we decide to go with the 'destroy in its own test' approach.

Reply:
From my side, I don't think that destroy-cluster should be in the test cases level at all as this is an infrastructure operation (logic wise).
In case someone wants to destroy it's cluster it should be a different phase in Jenkins and not part of the test suite